nmigen/hdl/ir.py

```diff
@@ -532,6 +532,13 @@ def prepare(self, ports=None, missing_domain=lambda name: ClockDomain(name)):
         if ports is None:
             fragment._propagate_ports(ports=(), all_undef_as_ports=True)
         else:
+            # typecheck ports is either list or tuple (#362)
+            if not isinstance(ports, tuple) and not isinstance(ports, list):
+                msg = "`ports` must be either a list or a tuple, not {!r}"\
+                        .format(type(ports).__name__)
+                if isinstance(ports, Value):
+                    msg += " (did you mean `ports=(<signal>,)`?)"
```

Comment:
This suggestion should be more specific, since a typo is often not obvious:
```suggestion
                    msg += " (did you mean `ports=(<signal>,)` rather than `ports=(<signal>)`?)"
```